R/cedta.R

```diff
@@ -37,7 +37,7 @@ cedta = function(n=2L) {
     (nsname %chin% cedta.pkgEvalsUserCode && any(sapply(sys.calls(), function(x) is.name(x[[1L]]) && (x[[1L]]=="eval" || x[[1L]]=="evalq")))) ||
     nsname %chin% cedta.override ||
     isTRUE(ns$.datatable.aware) ||  # As of Sep 2018: RCAS, caretEnsemble, dtplyr, rstanarm, rbokeh, CEMiTool, rqdatatable, RImmPort, BPRMeth, rlist
-    tryCatch("data.table" %chin% get(".Depends",paste("package",nsname,sep=":"),inherits=FALSE),error=function(e)FALSE)  # both ns$.Depends and get(.Depends,ns) are not sufficient
+    dependsNotImports(nsname)
```

Comment:
I would say it is risky due to possible overhead that would occur in many places. `.onLoad` could still provide a way for some validation on that without putting an overhead in a critical place.

Reply:
Won't this line of code only be run if the preceding are all `FALSE`? My first reaction was the same but I thought it is rare to get there

Reply:
Yes it only reaches here via the tests that call `base`, `utils` or `stats`. I've seen covr/codecov tell us the number of hits per line but I haven't seen it tell us where those hits came from.